tests/framework/common.go

```diff
@@ -1031,6 +1036,45 @@ func (td *OsmTestData) WaitForPodsRunningReady(ns string, timeout time.Duration,
 	return fmt.Errorf("not all pods were Running & Ready in NS %s after %v", ns, timeout)
 }
 
+// waitPodsReady waits for pods fetched by app label in a namespace to be ready in N seconds of time
+func (td *OsmTestData) waitPodsReady(ns string, appLabels []string, timeout time.Duration) bool {
```

Comment:
there's already a function `WaitForPodsRunningReady(...)` that ensures all pods in the control plane namespace are ready. Let's reuse that and integrate that check for all installs, and not just for `cert-manager`, `vault` scenarios.

Reply:
the existing impl checks for a number of ready pod running on a namespace, it's more basic in that sense. I anything, this is better as allows checking specific applications running as opposed to just a number of pods running (which I think is needed).
what would you suggest?

Reply:
Can we instead enhance that and wait for all pods in that namespace to be running? What's the benefit to only waiting for a subset of pods in the namespace?

Reply:
that was an easy way to wait for pods without going into the gory details as of how to do it, as we generally anyway we don't have bags of mixed stuff in test namespaces.
I'm not sure all uses of the previous impl would be able to migrate to the new one as it uses app labels (do single pod tests / no deployment tests use app label?)
I'll take a look regardless. I purposefully tried to avoid changing a broadly used api for the reasons above.

Reply:
My suggestion is to collapse all checks to a single API that checks if all pods in the control plane namespace are running.

Reply:
@shashankram take a look again. I augmented the available API to optionally take selectors. lmk